Source/WebService/Blackjack.WebService/ClientApp/src/app/components/play-game-page/play-game.component.ts

```diff
@@ -203,30 +203,32 @@ export class PlayGameComponent implements OnInit {
   /**
    * Resolves any situation in which the player wins a hand of blackjack
    */
-  public winOutcome(): void {
+  public async winOutcome(): Promise<void> {
     this.user.balance += (this.wager * 2);
     this.wager = 0;
-    this._userApi.updateUser(this.user);
+    await this._userApi.updateUser(this.user);
   }
 
   /**
    * Resolves any situation in which the player loses a hand of blackjack
-   */  public loseOutcome(): void {
+   */
+    public async loseOutcome(): Promise<void> {
     this.wager = 0;
-    this._userApi.updateUser(this.user);
+    await this._userApi.updateUser(this.user);
   }
 
   /**
    * Resolves any situation in which the player ties in a hand of blackjack
    */
-  public pushOutcome(): void {
+  public async pushOutcome(): Promise<void> {
     this.user.balance += this.wager;
     this.wager = 0;
+    await this._userApi.updateUser(this.user);    
   }
 
   public async ngOnInit(): Promise<void> {
```

Comment:
ngOnInit is the very first function that's ever run in your component.

It's usually a place to do various setup work like you're doing here.

For this reason it's common to put it at the top of the list of functions in the file just below the constructor.

This helps to make it obvious what lifecycle hooks are running and what they're doing.

Relocate this to the top of your list of functions just below the controller.